### PR TITLE
fix: qemu-guest-agent not starting in maintenance mode.

### PR DESCRIPTION
**Pull request by mlinares1998**
Allows qemu-ga to run under maintenance mode.

Resolves: #293

### CODE REVIEW

**Review by frezbo — APPROVED** [review]
:cool:

File: guest-agents/qemu-guest-agent/qemu-guest-agent.yaml [hunk 1]
```diff
@@ -20,7 +19,7 @@ container:
         - bind
         - ro
     # State files.
-    - source: /var/run
+    - source: /system/run/qemu-guest-agent
```
Comment (smira) [inline guest-agents/qemu-guest-agent/qemu-guest-agent.yaml:22]: does it need any state to be actually mounted from the host? given it's tmpfs, it just preserves state across service restarts, not host restarts (?)

Comment (frezbo) [reply guest-agents/qemu-guest-agent/qemu-guest-agent.yaml:22]: i don't think so, just needs a writable `/var/run`

Comment (smira) [reply guest-agents/qemu-guest-agent/qemu-guest-agent.yaml:22]: I just wonder if we need support for `emptyDir` (or some equivalent of it) mounts...

Comment (frezbo) [reply guest-agents/qemu-guest-agent/qemu-guest-agent.yaml:22]: this might fit well into our volumes ideas

Comment (smira) [reply guest-agents/qemu-guest-agent/qemu-guest-agent.yaml:22]: well, in this case it's per-container (or even per-container run) temporary mount, we could do same thing as kubelet does - create some temporary directory under `/system`  or `/tmp`, mount it up into the container, and remove the directory when the container stops?

Comment (smira) [reply guest-agents/qemu-guest-agent/qemu-guest-agent.yaml:22]: on other hand, it probably makes sense to backport it _this way_ to 1.6.x, and we can work on `emptyDir` mounts next

Comment (frezbo) [reply guest-agents/qemu-guest-agent/qemu-guest-agent.yaml:22]: > on other hand, it probably makes sense to backport it this way to 1.6.x, and we can work on emptyDir mounts next

yeh, let's get this merged in as and once we have emptyDir support, let's update

**Pull request merged**